### ISSUE TITLE
Use `pooch` for downloading the model:

### ISSUE THREAD

**Issue by GenevieveBuckley** [computational-cell-analytics/micro-sam#36]
> We could also think about using `pooch` for downloading the model: https://github.com/computational-cell-analytics/micro-sam/blob/master/micro_sam/util.py#L39-L63

_Originally posted by @constantinpape in https://github.com/computational-cell-analytics/micro-sam/issues/35#issuecomment-1573316601_

**Comment by hmaarrfk** [comment]
FYI, i've found that SHA256 is quite slow at computing hashes for large files. (i.e. 2GB models...)

you might want to play around with maybe cryptographically non-secure hashes. I've used XXH128 with great success.

**Comment by constantinpape** [comment]
Hi @hmaarrfk,

I think that would make sense. I don't think we require the hashing to be secure since the goal here is not to avoid malicious attacks, but just to make sure that files do not change.

In case you want to look into this: contributions are very welcome. (If not: no worries, but this would be on my list of lower priorities, so I can't really promise when we'd have time to look into this).
For details on how to contribute see https://github.com/computational-cell-analytics/micro-sam/blob/master/doc/contributing.md.

**Comment by GenevieveBuckley** [comment]
That should be straightforward to do, pooch supports the [python-xxhash](https://github.com/ifduyue/python-xxhash) bindings. 

The docs even have this nice example about how to use it, you just append `xxh128:` to the start of the hash.
https://www.fatiando.org/pooch/latest/hashes.html#other-supported-hashes

```python
import datetime
import pooch

# Get the current data to store the files in separate folders
CURRENT_DATE = datetime.datetime.now().date()

GOODBOY = pooch.create(
    [...],
    registry={
        "store.zip": "xxh128:6a71973c93eac6c8839ce751ce10ae48",
    },
)
```

**Comment by constantinpape** [comment]
Implemented in #276
